hail/python/hail/expr/expressions/typed_expressions.py

```diff
@@ -1472,8 +1472,15 @@ def __getitem__(self, item):
         """
         if isinstance(item, str):
             return self._get_field(item)
-        else:
+        if isinstance(item, int):
             return self._get_field(self.dtype.fields[item])
+        else:
+            assert item.start is None or isinstance(item.start, int)
+            assert item.stop is None or isinstance(item.stop, int)
+            assert item.step is None or isinstance(item.step, int)
+            return hl.struct(**{
```

Comment:
I think this should be a `select`, not `hl.struct`. The missingness semantics are different.

Reply:
And this is why I really like getting your feedback on python interface changes.